packages/kbn-type-summarizer/src/lib/ts_nodes/dec_symbol.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import ts from 'typescript';
+
+export type DecSymbol = ts.Symbol & {
+  declarations: NonNullable<ts.Symbol['declarations']>;
+};
+
+export function isDecSymbol(symbol: ts.Symbol): symbol is DecSymbol {
+  return !!symbol.declarations && symbol.declarations.length > 0;
+}
+
+export function assertDecSymbol(symbol: ts.Symbol): asserts symbol is DecSymbol {
+  if (!isDecSymbol(symbol)) {
+    throw new Error(
+      // TODO: expand this error message, maybe ignore symbols which resolve to non DecSymbols
```

Comment:
When we were walking through the code on zoom we talked about improving this error message with useful . We probably should at list trying to mention what is the line causing the error

Reply:
I actually ended up solving this a slightly different way so that we could have information about the code leading up to this error. The error message I had in place was focused around imports which don't resolve to node_modules, but it also applies to this case where someone is exporting JS from their package but instead of defining basic types in a `.d.ts` file or something they just use a `// @ts-expect-error` directive or something. Expanded the error message for that case and added a test for it in [`0eac579` (#135163)](https://github.com/elastic/kibana/pull/135163/commits/0eac5795ca69190d9d5acc5af6a814fe0a1090f5)